### DISCUSSION TITLE
Issue: 401 Error When Using Active API Key with Transitland REST API

### DISCUSSION THREAD

**Discussion by KeXu98619** [transitland/transitland#439] [Q&A] [answered]
Hi, I’ve registered for a free Interline account and received an active API key. However, when trying to use the REST API, I consistently receive a "401: No API key found in request" error. I’ve tested my key with a simple /status endpoint, but the issue persists.

Here’s the code I’m using:
```
import requests

API_KEY = "REDACTED"
url = "https://transit.land/api/v2/rest/status"
headers = {
    "Authorization": f"Bearer {API_KEY}",
    "Accept": "application/json"
}
response = requests.get(url, headers=headers)
print(response.status_code)
print(response.json())

```

I've included my API key above. Can someone help me look into this? Thanks!

**Comment by drewda** [comment, marked as answer]
We've removed your API key from the above question.

Please note that there is no https://transit.land/api/v2/rest/status endpoint Instead try an endpoint such as https://transit.land/api/v2/rest/stops

Your API key can be provided as a query parameter or a header using one of the keys listed as: https://www.transit.land/documentation/index#signing-up-for-an-api-key

So please try revising your code to:

```py
import requests

API_KEY = "REDACTED"
url = "https://transit.land/api/v2/rest/stops"
headers = {
    "apikey": API_KEY
    "Accept": "application/json"
}
response = requests.get(url, headers=headers)
print(response.status_code)
print(response.json())

```

**Reply by KeXu98619** [reply]
Thanks for your help, it works after I change the apikey as a query parameter.
